pynxtools_mpes/reader.py

```diff
@@ -285,128 +164,98 @@ def _getattr(obj, attr):
     return reduce(_getattr, [obj] + attr.split("."))
 
 
-def fill_data_indices_in_config(config_file_dict, x_array_loaded):
-    """Add data indices key value pairs to the config_file
-    dictionary from the xarray dimensions if not already
-    present.
-    """
-    for key in list(config_file_dict):
-        if "*" in key:
-            value = config_file_dict[key]
-            for dim in x_array_loaded.dims:
-                new_key = key.replace("*", dim)
-                new_value = value.replace("*", dim)
-
-                if (
-                    new_key not in config_file_dict.keys()
-                    and new_value not in config_file_dict.values()
-                ):
-                    config_file_dict[new_key] = new_value
-
-            config_file_dict.pop(key)
+class MPESReader(MultiFormatReader):
+    """MPES-specific reader class"""
 
+    supported_nxdls = ["NXmpes", "NXmpes_arpes"]
+    config_file: Optional[str] = None
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.eln_data = None
+
+        self.extensions = {
+            ".yml": self.handle_eln_file,
+            ".yaml": self.handle_eln_file,
+            ".json": self.set_config_file,
+            ".h5": self.handle_hdf5_file,
+            ".hdf5": self.handle_hdf5_file,
+        }
+
+    def handle_hdf5_file(self, file_path: str) -> Dict[str, Any]:
+        """Handle hdf5 file"""
+        self.data_xarray = h5_to_xarray(file_path)
+
+        return {}
+
+    def set_config_file(self, file_path: str) -> Dict[str, Any]:
+        if self.config_file is not None:
+            logger.info(f"Config file already set. Skipping the new file {file_path}.")
+        self.config_file = file_path
+        return {}
+
+    def handle_eln_file(self, file_path: str) -> Dict[str, Any]:
+        self.eln_data = parse_yml(
+            file_path,
+            convert_dict=CONVERT_DICT,
+            replace_nested=REPLACE_NESTED,
+            parent_key="/ENTRY",
+        )
+
+        return {}
+
+    def get_eln_data(self, key: str, path: str) -> Any:
+        """Returns data from the given eln path."""
+        if self.eln_data is None:
+            return None
+
+        return self.eln_data.get(path)
```

Comment:
I am pretty sure that wouldn't work. You would probably like to have something like

```suggestion
        return self.eln_data.get(key.replace(f"/ENTRY[{self.callbacks.entry_name}]", "/ENTRY"))
```

Reply:
Ah yes. Thanks. We don't really use the `@eln` so it slipped. Generally, I think it would be good to have a path which does not depend on the actual key (so we don't actually have to flatten and replace the nomad eln) but there rather exists an eln path from which data may be retrieved.

Anyways, for now thanks for the fix. I put in.

Reply:
Yeah, I agree. There is actually no point (also on the xps side) to use the `convert_dict`. But for now, let's keep it as is.